CDCgov/NEDSS-Modernization
# Routing pb manage pages

## Description

When clicking the `Manage Pages` link in the Classic™ System management screen (see below), the user should be sent to the modernized Page Library screen.

![image](https://github.com/CDCgov/NEDSS-Modernization/assets/109251240/0fc3f59d-096c-4b3a-947c-be438265fca1)

## Authentication Filter Changes
This PR also combines the `ClassicIncomingAuthenticationFilter` and the `JWTFilter` so the application can handle a `JWT` or `JSESSIONID` cookie for any request.

For the new filter, the presence of a JWT is checked first. If the token is valid, the security context is updated and the next filter called. 
If the JWT is invalid, a redirect to timeout is issued.
If the JWT is Expired or doesn't exist, authentication using the JSESSIONID is attempted.

## Tickets

* [CNFT2-1223](https://cdc-nbs.atlassian.net/browse/CNFT2-1223)

## Steps to verify

1. Rebuild modernization-api and nbs-gateway projects
2. Navigate to http://localhost:8080/nbs/login
3. Clear all cookies
4. Login as `superuser`
5. Click `System Management`
6. Click `Manage Pages`
7. Verify Modernized Page Library screen is loaded

![manage-page-redirect](https://github.com/CDCgov/NEDSS-Modernization/assets/109251240/b3a4ab0d-10fa-4d90-9838-eda0ff6dac1d)


[CNFT2-1223]: https://cdc-nbs.atlassian.net/browse/CNFT2-1223?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQb
